src/DurableTask.Netherite/PartitionState/InstanceState.cs

```diff
@@ -39,12 +42,19 @@ public override string ToString()
 
         public override void Process(CreationRequestReceived creationRequestReceived, EffectTracker effects)
         {
+            if (creationRequestReceived.EventIdString == this.CreationRequestEventId)
```

Comment:
I wonder if there's a way to deduplicate EH earlier in the pipeline, so that we don't have to sprinkle de-duplication logic in various places. Not sure if there's some kind of counter we can use to determine that we've processed all events up to some ID, or something like that. Hard to say, I could also see the possibility that we need to handle each case separately.

Reply:
I don't think there is anything we can do here, because we cannot keep track of the lifetime of clients.